cluster/hooks.go

```diff
@@ -246,19 +240,48 @@ func InstallLogging(input interface{}, param pkgCluster.PostHookParam) error {
 	if err != nil {
 		return err
 	}
-	loggingValues := map[string]interface{}{
-		"bucketName": loggingParam.BucketName,
-		"region":     loggingParam.Region,
-		"secret": map[string]interface{}{
-			"secretName": installedSecretValues.Name,
-		},
-	}
-	marshaledValues, err := yaml.Marshal(loggingValues)
+
+	// Determine the type of output plugin
+	logSecret, err := secret.Store.Get(cluster.GetOrganizationId(), loggingParam.SecretId)
 	if err != nil {
 		return err
 	}
+	switch logSecret.Type {
+	case pkgCluster.Amazon:
+		installedSecretValues, err := InstallSecretWithVaultID(cluster, loggingParam.SecretId, loggingParam.GenTLSForLogging.Namespace)
```

Comment:
The `InstallSecretWithVaultID` is not idempotent thus the hook may fail in case of a re-run. Please consider using/addin a version of `InstallSecretWithVaultID` (see as an example `InstallOrUpdateSecrets`)